index.mjs

```diff
@@ -1,10 +1,11 @@
+const midCache = new Map(); // midpoint vertices cache to avoid duplicating shared vertices
```

Comment:
Initially I did it this way too, but eventually decided against this because of the way this module is used — it's unlikely to repeatedly call `icosphere` within the same app (usually you just generate one mesh, and use instanced rendering if you need multiple spheres), and in that case it would be better to have the module free up the memory used for midpoints immediately after the mesh has been retrieved.